pep-0XXX-environment-markers.rst

```diff
@@ -0,0 +1,174 @@
+difficult, as the set of required packages differs depending on the
+version of Python in use.
+
+For other packages, this means broad cross-platform support. However,
+this may mean (for instance) that certain packages are required for
+installation on Windows, but not other platforms.
+
+Environment Markers attempt to provide more flexibility in a list of
+requirements by allowing the developer to give some information about
+the conditions under which a requirement should be installed.
+
+
+Examples
+========
+
+Here are some examples of such markers inside a requirements.txt::
+
+   "sys_platform == 'win32'"
+   "platform_machine == 'i386'"
+   "python_version == '2.4' or python_version == '2.5'"
+   "python_version>='2.7' and python_version<'3.0'"
+   "'linux' in sys_platform"
+
+And here's an example of some conditional metadata included in
+setup.py for a distribution that requires PyWin32 both at runtime and
+buildtime when using Windows::
+
+   setup(
+     install_requires=["pywin32 > 1.0 : sys.platform == 'win32'"],
+     setup_requires=["pywin32 > 1.0 : sys.platform == 'win32'"]
+     )
+
+
+Micro-language
+==============
+
+The micro-language behind this is as follows. It compares::
+
+* strings with the ``==`` and ``in`` operators (and their opposites)
+* version numbers with the ``<``, ``<=``, ``>=``, and ``<`` operators
+  in addition to those supported for strings
+
+The usual boolean operators ``and`` and ``or`` can be used to combine
+expressions, and parentheses are supported for grouping.
+
+The pseudo-grammar is ::
+
+    MARKER: EXPR [(and|or) EXPR]*
+    EXPR: ("(" MARKER ")") | (STREXPR|VEREXPR)
+    STREXPR: STRING [STRCMPOP STREXPR]
+    STRCMPOP: ==|!=|in|not in
+    VEREXPR: VERSION [VERCMPOP VEREXPR]
+    VERCMPOP: (==|!=|<|>|<=|>=)
+
+
+``SUBEXPR`` is either a Python string (such as ``'win32'``) or one of
+the ``Strings`` marker variables listed below.
+
+``VEREXPR`` is a PEP-0440[3] version identifier, or one of the
+``Version number`` marker variables listed below. Comparisons between
+version numbers are done using PEP-0440 semantics.
+
+Version numbers
+---------------
+
+* ``python_version``: ``'{0.major}.{0.minor}'.format(sys.version_info)``
+* ``python_full_version``: see definition below
+* ``platform_version``: ``platform.version()``
+* ``implementation_version````: see definition below
+
+
+Strings
+-------
+
+* ``os_name````: ``os.name``
+* ``sys_platform````: ``sys.platform``
+* ``platform_release``: ``platform.release()``
+* ``implementation_name````: ``sys.implementation.name``
+* ``platform_machine``: ``platform.machine()``
+* ``platform_python_implementation``: ``platform.python_implementation()``
+
+
+If a particular string value is not available (such as ``sys.implementation.name``
+in versions of Python prior to 3.3), the corresponding marker
+variable MUST be considered equivalent to the empty string.
+
+If a particular version number value is not available (such as
```

Comment:
... here, such that the description of version variables comes entirely after the description of string variables.